src/pal/src/locale/utf8.cpp

```diff
@@ -1463,7 +1463,7 @@ class UTF8Encoding
                 }
 
                 // get pSrc 2-byte aligned
-                if (((uintptr_t)pSrc & 0x1) != 0) {
+                if (((intptr_t)pSrc & 0x1) != 0) {
```

Comment:
Can we mirror runtime here (just to make it easier for future changes)? They use `size_t`